src/QMCHamiltonians/QMCHamiltonian.h

```diff
@@ -466,6 +483,19 @@ class QMCHamiltonian
   // helper function for extracting a list of Hamiltonian components from a list of QMCHamiltonian::H.
   static RefVectorWithLeader<OperatorBase> extract_HC_list(const RefVectorWithLeader<QMCHamiltonian>& ham_list, int id);
 
+  //This is for fast derivative evaluation.  
+  TWFFastDerivWrapper psi_wrapper_;
+  std::vector<ValueMatrix> X_;    //Working arrays for derivatives
+  std::vector<ValueMatrix> Minv_; //Working array for derivatives.
+  std::vector<ValueMatrix> B_;
+  std::vector<ValueMatrix> B_gs_;
+  std::vector<ValueMatrix> M_;
+  std::vector<ValueMatrix> M_gs_;
+
+  std::vector<std::vector<ValueMatrix>> dM_;
+  std::vector<std::vector<ValueMatrix>> dM_gs_;
+  std::vector<std::vector<ValueMatrix>> dB_;
+  std::vector<std::vector<ValueMatrix>> dB_gs_;
```

Comment:
Are all these storage needed by one function?
Could you create a struct and make the ACforce owning a member of this struct and pass the struct by reference via the QMCHamiltonian force API?
The intention is to keep QMCHamiltonian more of an interface class and avoid implementation details.

Reply:
These are just intermediates required by QMCHamiltonian::evaluateIonDerivsDeterministicFast(...).  I agree that these shouldn't be member variables.  If it's OK with you, I might just have these declared and inititalized entirely within QMCHamiltonian::evaluateIonDerivsDeterministicFast()...  Not optimal, but memory allocation is such a tiny fraction of the total time for this function that it's probably OK for now.  And then we don't have to worry about adding structs.

Reply:
Yes. Do it in evaluateIonDerivsDeterministicFast